include/glow/Backends/CompiledFunction.h

```diff
@@ -38,21 +38,21 @@ class CompiledFunction {
   virtual ~CompiledFunction() = default;
   /// Execute the network and allocate Placeholder memory with given
   /// \p ctx providing mapping between Placeholder and populated tensor.
-  virtual void execute() = 0;
+  virtual void execute(Context *) = 0;
 
-  /// Does any needed initialization work for the Backend.
+  /// DEPRECATED. Does any needed initialization work for the Backend.
   /// This includes device init constant memory allocation and copying to
   /// device.
-  virtual void setupRuns() = 0;
+  virtual void setupRuns() { runsSetup_ = true; }
 
-  /// Per run setup. Copy inputs to device.
-  virtual void beforeRun(const Context &ctx) = 0;
+  /// DEPRECATED. Per run setup. Copy inputs to device.
+  virtual void beforeRun(const Context &ctx) {}
 
-  /// Per run cleanup. Copy outputs from device.
-  virtual void afterRun(const Context &ctx) = 0;
+  /// DEPRECATED. Per run cleanup. Copy outputs from device.
+  virtual void afterRun(const Context &ctx) {}
 
-  /// Final cleanup. Release memory, reset device.
-  virtual void tearDownRuns() = 0;
+  /// DEPRECATED> Final cleanup. Release memory, reset device.
+  virtual void tearDownRuns() { runsSetup_ = false; }
```

Comment:
the comment below says: /// Flag to ensure setupRuns is only called once.

is it still the case, if so why we flip it back to false.

Reply:
@nickgg You can use the deprecated attribute, this will issue a warning:

https://clang.llvm.org/docs/AttributeReference.html#deprecated

Reply:
@rdzhabarov This is how the existing code works, I suppose the idea is that the lifetime of a "Run" is different from the lifetime of the CompiledFunction. Ultimately the best solution is to remove these functions entirely once all device types are updated.

Reply:
@nadavrot Is a warning what we want now since we haven't updated the CompiledFunctions for all devices yet?